include/DeterministicNoiseSimulator.hpp

```diff
@@ -3,124 +3,132 @@
+                                double                                   cNoiseProbability,
+                                std::optional<double>                    cAmpDampingProbability,
+                                double                                   cMultiQubitGateFactor,
+                                bool                                     unoptimizedSim = false,
+                                unsigned long long                       seed           = 0):
+        Simulator<DDPackage>(seed),
+        qc(qc),
+        noiseProbSingleQubit(cNoiseProbability),
+        noiseProbMultiQubit(cNoiseProbability * cMultiQubitGateFactor),
+        sequentiallyApplyNoise(unoptimizedSim),
+        useDensityMatrixType(!unoptimizedSim) {
+        // setNoiseEffects
+        for (auto noise: cNoiseEffects) {
+            switch (noise) {
+                case 'A':
+                    noiseEffects.push_back(dd::amplitudeDamping);
+                    break;
+                case 'P':
+                    noiseEffects.push_back(dd::phaseFlip);
+                    break;
+                case 'D':
+                    noiseEffects.push_back(dd::depolarization);
+                    break;
+                case 'I':
+                    noiseEffects.push_back(dd::identity);
+                    break;
+                default:
+                    throw std::runtime_error("Unknown noise operation '" + cNoiseEffects + "'\n");
+            }
+        }
 
-    void initializeNoiseProbabilities(double cGateNoiseProbability, double cAmplitudeDampingProb = -1, double cMultiQubitGateFactor = 2) {
-        noiseProb            = cGateNoiseProbability;
-        noiseProbSingleQubit = cGateNoiseProbability;
-        noiseProbMultiQubit  = cGateNoiseProbability * cMultiQubitGateFactor;
+        // initializeNoiseProbabilities
+        noiseProbSingleQubit = cNoiseProbability;
+        noiseProbMultiQubit  = cNoiseProbability * cMultiQubitGateFactor;
 
-        if (cAmplitudeDampingProb < 0) {
+        if (!cAmpDampingProbability) {
             // Default value for amplitude damping prob is double the general error probability
-            ampDampingProb            = cGateNoiseProbability * 2;
-            ampDampingProbSingleQubit = cGateNoiseProbability * 2;
-            ampDampingProbMultiQubit  = cGateNoiseProbability * 2 * cMultiQubitGateFactor;
+            ampDampingProbSingleQubit = cNoiseProbability * 2;
+            ampDampingProbMultiQubit  = cNoiseProbability * 2 * cMultiQubitGateFactor;
         } else {
-            ampDampingProb            = cAmplitudeDampingProb;
-            ampDampingProbSingleQubit = cAmplitudeDampingProb;
-            ampDampingProbMultiQubit  = cAmplitudeDampingProb * cMultiQubitGateFactor;
+            ampDampingProbSingleQubit = cAmpDampingProbability.value();
+            ampDampingProbMultiQubit  = cAmpDampingProbability.value() * cMultiQubitGateFactor;
         }
```

Comment:
Isn't this computation rather redundant with the computations in the noise functionality that now resides in the QFR?

Reply:
Computations are not repeated within the QFR. The values are initialized here, since the sanity checks are also done here.